interactions/api/models/guild.py

```diff
@@ -1638,14 +1638,21 @@ async def get_role(
         :rtype: Role
         """
 
-        if not self._client:
-            raise LibraryException(code=13)
-        roles = await self._client.get_all_roles(guild_id=int(self.id))
-        for i in roles:
-            if int(i["id"]) == role_id:
-                role = Role(**i)
-                break
-        return role
+        try:
+            role = next(role for role in self.roles if int(role.id) == role_id)
+            return role
+        except StopIteration:  # sourcery skip: raise-from-previous-error
+            if not self._client:
+                raise LibraryException(code=13)
+            roles = await self._client.get_all_roles(guild_id=int(self.id))
+            try:
+                role = next(role for role in roles if int(role["id"]) == role_id)
+                self.roles = [Role(**_) for _ in roles]
+                return Role(**role)
+            except StopIteration as exc:
+                raise LibraryException(
+                    message="The role you looked for was not found!", code=0, severity=30
+                ) from exc
```

Comment:
```suggestion
        for role in self.roles:
            if int(role.id) == role_id:
                return role
        else:
            if not self._client:
                raise LibraryException(code=13)
            roles = await self._client.get_all_roles(guild_id=int(self.id))
            self.roles = [Role(**_) for _ in roles]
            for role in self.roles:
                if int(role.id) == role_id:
                    break
            else:
                raise LibraryException(
                    message="The role you looked for was not found!", code=0, severity=30
                )
            return role
```
Use for else
DISCLAIMER: did this on phone, pls test and format the code if necessary

Reply:
What do you have against `next()`?

Reply:
Next loops till end of list
Breaking early in for loops is better
Also no catching exceptions, else is used instead

Reply:
Is for/else even a thing?

Reply:
Also the code is shorter

Reply:
Yes

Reply:
Also please remove the `from exc` from the pre-last line

Reply:
Else is run if you don't break from for loop

Reply:
Ok

Reply:
> Also please remove the `from exc` from the pre-last line

Done

Reply:
Alright nice